api/src/main/java/org/openmrs/validator/ConditionValidator.java

```diff
@@ -50,5 +52,8 @@ public void validate(Object object, Errors errors) {
 		if(condition.getClinicalStatus() == null){
 			errors.rejectValue("clinicalStatus", "Condition.clinicalStatusShouldNotBeNull", "The clinical status is required");
 		}
+		if(condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) && condition.getOnsetDate().compareTo(condition.getEndDate()) > 0){
+			errors.rejectValue("onsetDate","Condition.onsetDateShouldBeBeforeEndDate","Valid Onset Date is required");
+		}
```

Comment:
This would work:
```java
if (condition.getClinicalStatus() == null){
  errors.rejectValue("clinicalStatus", "Condition.clinicalStatusShouldNotBeNull", "The clinical status is required");
}
else if (condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) && condition.getOnsetDate().compareTo(condition.getEndDate()) > 0) {
  errors.rejectValue("onsetDate","Condition.onsetDateShouldBeBeforeEndDate","Valid Onset Date is required");
}
```

Reply:
Thank you @mks-d . This fixed the issue :)